packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -94,13 +94,17 @@ export default createRule<Options, MessageIds>({
       if (!valueType) {
         return;
       }
-
+      const value = sourceCode.getText(valueType.typeAnnotation);
+      const name = node.parent?.id?.name;
+      const valueArray = value.split('|').map(v => v.trim());
+      if (valueArray.indexOf(name) != -1) {
```

Comment:
doing operations and linting based on text is not correct and will always cause false positive or false negative.

here are a number of cases you'll false-positive
```ts
type T = { [k: string]: AT };
type T = { [k: string]: TA };
type T = { [k: string]: A.T };
type T = { [k: string]: A['T'] };
type T = { [k: string]: (A: string) => void };
// ...
```

You need to use scope analysis here.
https://eslint.org/docs/developer-guide/scope-manager-interface

Look for references within the current node.
search this or the ESLint core codebase for `context.getScope` for examples

Reply:
Okay, I'll come back with a fix as soon as possible, thanks for the help.

Reply:
@bradzacher SO I have to use eslint-scope dependency?

Reply:
@bradzacher  I did some changes but I feel the logic I implemented is too bloated, please review and let me know if I can refactor it, thankyou.